docs/INSTALLATION.md

```diff
@@ -1,65 +1,82 @@
-compiler) when it is required. You can start the installation and test
-by running the following command in a terminal:
+## Install Stack
 
-```bash
-stack ghc
-```
+For doing Exercism exercises you only need Stack.
+
+To install Stack, simply run the GHCup installation command or script and follow the defaults but specify that you would like to have Stack installed as well.
 
-If this outputs something like `ghc: no input files`, GHC is installed
-correctly.
+Did you miss the opportunity? No worries: just run `ghcup install stack` afterwards.
 
-Note that there are other ways to install GHC on a system (like the
-[Haskell platform](https://www.haskell.org/platform/)), but for
-running the Exercism tests, a regular stack installation is most
-likely to work correctly.
+GHCup will install
 
-## Installing HLint (optional)
+- **A compiler** (GHC: the Glasgow Haskell Compiler)
+  
+  Strictly speaking you do not need this for Exercism, but it comes in handy when you want to play with GHCi outside of exercise directories.
 
-HLint is a tool for suggesting possible improvements to Haskell code. These
-suggestions include ideas such as using alternative functions, simplifying
-code and spotting redundancies.
+- **Cabal**
 
-You can use Stack to install HLint:
+  A project/package manager. You do not need this for Exercism.
+
+- **Stack**
+
+  A project/package manager that you do need for Exercism. It will make sure you have the right compiler and libraries for each individual project (exercise).
+  
+To check that installation was successful, run the command `ghcup list` and/or `stack --version`.
+
+
+## Get the right GHC version
+
+Stack will automatically download and install GHC when it is required. You can start the installation and test
+by running the following command in a terminal:
 
 ```bash
-stack install hlint
+# move into exercise directory
+cd /exercism/haskell/hello-world
+# start interactive environment
+stack ghc -- --version
 ```
 
-Check that it was installed correctly with:
+If this outputs something like
 
-```bash
-hlint --version
 ```
+The Glorious Glasgow Haskell Compilation System, version 8.8.4
+```
+
+then GHC is installed correctly.
+
+
+## Set up your editor to work with Haskell
+
+Haskell support is provided by the Haskell Language Server (HLS).
+
+
+### Visual Studio Code
+
+Just get [this extension][vscode-haskell]. It will use GHCup to install HLS automatically, after which you'll be all set.
+
+**Note:** with freshly downloaded exercises, HLS may have trouble starting up. If this happens, just run `stack test` and restart the language server.
+
 
-If you get the version in your terminal, you're done, otherwise, you
-probably need to add a missing directory to your path:
+### Other editors
 
-- `%APPDATA%\local\bin`, if you are running *Windows*
-- `$HOME/.local/bin`, for other operating systems.
+See the [HLS documentation on editor configuration][hls-editor-config].
```

Comment:
I'm conflicted. On the one hand, the student may get confused about "what is Haskell Language Server and do I need this?". See how a lot of the install instructions also include instructions to install a language server protocol implementation, which is not in line with the stated purpose of this link (other editors). On the other hand, I wanted to suggest https://wiki.haskell.org/Editors, but that provides no commentary about each editor. Thus I don't think either of these suggestions is perfect. I would like to figure out which will be most helpful to the students.